src/psyclone/psyir/transformations/hoist_local_arrays_trans.py

```diff
@@ -168,12 +168,12 @@ def apply(self, node, options=None):
             # symbol name already present at container scope. (The validate()
             # method will already have checked for tag clashes.)
             try:
-                container.symbol_table.add(sym, tag=tags_dict.get(sym))
+                container.symbol_table.add(sym, tag=tags_dict.get(sym.name))
```

Comment:
This now provides the wrong tag for:
```
Routine with SymbolTable [symbol1("a", tag="a")]
  IfStmt
     Schedule with SymbolTable [symbol2("a", tag="b")]
```
As symbol1 and symbol2 will return the same for `tags_dict.get(sym.name)` but clearly one of them is wrong.

Also instead of `symtab.get_reverse_tags_dict()[symbol]` I would prefer `symtab.lookup_symbol_tags(symbol)`:
```
def lookup_symbol_tags(self, symbol):  
    current = self
    while current:
        for check_symbol in current.symbols_dict.values():
            if symbol == check_symbol:  
                return current.tags_dict(symbol.name)
        current = current.parent_symbol_table()
    return None
```
and get rid of the get_reverse_tags_dict

Reply:
Reverted to using `sym` now.